src/systems/filecoin_mining/storage_mining/storage_miner_actor_state.go

```diff
@@ -115,17 +115,37 @@ func MinerInfo_New(
 	ownerAddr addr.Address, workerAddr addr.Address, sectorSize sector.SectorSize, peerId libp2p.PeerID) MinerInfo {
 
 	ret := &MinerInfo_I{
-		Owner_:      ownerAddr,
-		Worker_:     workerAddr,
-		PeerId_:     peerId,
-		SectorSize_: sectorSize,
+		Owner_:            ownerAddr,
+		_worker_:          workerAddr,
+		PeerId_:           peerId,
+		SectorSize_:       sectorSize,
+		PendingKeyChange_: nil,
 	}
 
 	TODO() // TODO: determine how to generate/validate VRF key and initialize other fields
 
 	return ret
 }
 
+func (mi *MinerInfo_I) GetWorker(currEpoch block.ChainEpoch) addr.Address {
+	if mi.PendingKeyChange() == nil {
+		return mi._worker()
+	}
+
+	// lazy updates to worker keys if at effectiveness
+	if mi.PendingKeyChange().EffectiveAt() <= currEpoch {
+		mi._worker_ = mi.PendingKeyChange().NewKey()
+		mi.PendingKeyChange_ = nil
+		util.TODO()
+		// remains to update the key in every deal, or associate deals to minerID
+	}
+	return mi._worker()
+}
```

Comment:
We can't do state updates here -- it's used as a pure function elsewhere (e.g., in `TerminateSector`), where we call `Release` instead of `UpdateRelease`. Let's split this stanza out into a separate function, if we need to do this lazy update. (It's not clear to me that we do, though -- isn't the logic identical if we just return the would-be updated worker here, without actually committing any state updates?)

Reply:
Agreed we can't do state updates here, and I think the silent nature is confusing. We should make this change quite explicit.

My preferred approach would be to add a CommitKeyChange explicit method which will persist the change if called after suitable delay

Reply:
Changed accordingly.

A question though, if the `CommitKeyChange()` method is called explicitly, then there may be some number of epochs in which the key could be changed but is not changed yet, thereby allowing the storageMiner to "straddle" two worker keys momentarily and only make the switch when relevant. 

I claim that unless the systemActor basically queues up key changes to push them at the appropriate epoch with the cron tick, then the rational strategy is to always run two keys to gain an advantage. Do you agree @anorth @jzimmerman ?
If so, it's worth reworking this so the cron actor prompts the key change on time.

Reply:
Fair concern. I was hoping to avoid the cron because it'll involve more complexity. But I think in the giant actor PR of late December a job queue type thing was added to SPA, so this could register for a callback from that.